java/ql/lib/semmle/code/java/frameworks/apache/IOGenerated.qll

```diff
@@ -1,684 +1,828 @@
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(File);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(File,Charset);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(File,Charset,boolean,String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(File,String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(File,String,boolean,String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(File,boolean);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(File,boolean,String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(String,boolean);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(String,boolean,String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;XmlStreamWriter;true;XmlStreamWriter;(File);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;XmlStreamWriter;true;XmlStreamWriter;(File,String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.serialization;ClassNameMatcher;true;matches;(String);;Argument[-1];generated:regex-use[0]",
+        "org.apache.commons.io.serialization;ClassNameMatcher;true;matches;(String);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;CopyUtils;true;copy;(InputStream,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(InputStream,Writer,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(Reader,OutputStream);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(Reader,OutputStream,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(Reader,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(String,OutputStream);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(String,OutputStream,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(String,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(byte[],Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(byte[],Writer,String);;Argument[0];generated:write-file",
```

Comment:
These all look like they should be taint-prop steps arg0 -> arg1, plus the written arg is arg1 not arg0

Reply:
These were derived due to java.io.Writer handwritten write-file sink. I have removed this sink.